living_documentation_generator/generator.py

```diff
@@ -0,0 +1,316 @@
+
+
+class LivingDocumentationGenerator:
+
+    PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))
+
+    ISSUE_PAGE_TEMPLATE_FILE = os.path.join(PROJECT_ROOT, "..", "templates", "issue_detail_page_template.md")
+    INDEX_PAGE_TEMPLATE_FILE = os.path.join(PROJECT_ROOT, "..", "templates", "_index_page_template.md")
+
+    def __init__(self, repositories: list[ConfigRepository],
+                 projects_title_filter: list[str] = None,
+                 project_state_mining_enabled: bool = False,
+                 output_path: str = "../output"):
+
+        # data
+        self._repositories: list[ConfigRepository] = repositories
+        self._projects_title_filter: list[str] = projects_title_filter      # TODO - this could be part of repository config ???
+
+        # features control
+        self._project_state_mining_enabled: bool = project_state_mining_enabled
+
+        # paths
+        self._output_path: str = output_path
+
+    @property
+    def repositories(self) -> list[ConfigRepository]:
+        return self._repositories
+
+    @property
+    def projects_title_filter(self) -> list[str]:
+        return self._projects_title_filter
+
+    @property
+    def project_state_mining_enabled(self) -> bool:
+        return self._project_state_mining_enabled
+
+    @property
+    def output_path(self) -> str:
+        return self._output_path
+
+    def generate(self):
+        self._clean_output_directory()
+
+        # Data mine GitHub issues with defined labels from repository
+        gh_repo_issues: dict[str, list[Issue]] = self._mine_github_issues()
+        # Note: got dict of list of issues for each repository (key is repository id)
+
+        # Data mine GitHub project's issues
+        gh_project_issues = self._mine_github_project_issues()
+
+        # Consolidate all issues data together
+        projects_issues = self._consolidate_issues_data(gh_repo_issues, gh_project_issues)
+
+        # Generate markdown pages
+        self._generate_markdown_pages(projects_issues)
+
+    def _clean_output_directory(self):
+        if os.path.exists(self.output_path):
+            shutil.rmtree(self.output_path)
+        os.makedirs(self.output_path)
+
+    def _mine_github_issues(self) -> dict[str, list[Issue]]:
+        issues = {}
+
+        # Mine issues from every config repository
+        for config_repository in self.repositories:
+            repository_id = f"{config_repository.owner}/{config_repository.name}"
+
+            if GithubManager().store_repository(repository_id) is None:
+                return {}
+
+            logging.info(f"Downloading issues from repository `{config_repository.owner}/{config_repository.name}`.")
+
+            # Load all issues from one repository (unique for each repository)
+            repo_issues = GithubManager().fetch_issues(labels=config_repository.query_labels)
+            for repo_issue in repo_issues:
+                repo_issue.repository = repository_id
```

Comment:
This is unnecessary. We don't use repository_id later nowhere, because we build repository_id again in consolidation from ConfigRepo

Reply:
Nice. Addressed in commit - TBD.